### PR TITLE
[content-service] Refactor GCS test to use a fake server

### PR DESCRIPTION
**Pull request by aledbf**
## Description

Introduce `github.com/fsouza/fake-gcs-server/fakestorage` to mock GCS. No other change.
Additional tests will be added in a different PR

## Release Notes
```release-note
[content-service] Refactor GCS test to use a fake server
```

### CODE REVIEW

File: components/content-service/pkg/storage/gcloud_test.go [hunk 1]
```diff
@@ -6,22 +6,42 @@ package storage
 
 import (
 	"context"
-	"io"
 	"testing"
 
-	"golang.org/x/xerrors"
+	"cloud.google.com/go/storage"
+	"github.com/fsouza/fake-gcs-server/fakestorage"
```
Comment (utam0k) [inline components/content-service/pkg/storage/gcloud_test.go:12]: 👍

**Review by utam0k — APPROVED** [review]

File: components/content-service/pkg/storage/gcloud_test.go [hunk 2]
```diff
@@ -6,22 +6,42 @@ package storage
...
+		option.WithEndpoint(server.URL()),
+		option.WithHTTPClient(server.HTTPClient()),
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+
+	client, err := storage.NewClient(ctx, args...)
+	if err != nil {
+		t.Errorf("error creating GCS gcsClient: %v", err)
```
Comment (jenting) [inline components/content-service/pkg/storage/gcloud_test.go:34]: nit
```suggestion
                t.Errorf("error creating GCS client: %v", err)
```

**Review by jenting — APPROVED** [review]
LGTM

```shell
go test -timeout 30s -run ^TestObjectAccessToNonExistentObj$ github.com/gitpod-io/gitpod/content-service/pkg/storage

ok  	github.com/gitpod-io/gitpod/content-service/pkg/storage	0.009s
```

**Pull request closed without merging**